projects/noloan/app/src/main/java/hasadna/noloan/DbMessages.java

```diff
@@ -99,52 +152,75 @@ public int getIndexById(SmsMessage smsMessage) {
     return -1;
   }
 
-  // region Functions used by db Listeners when list changes
+  public FirebaseUser getFirebaseUser() {
+    return firebaseUser;
+  }
+
+  // Receive changes from the db (called from FirestoreClient)
+  public void updateChange(SmsMessage smsMessage, Type type) {
+    switch (type) {
+      case ADDED:
+        addMessage(smsMessage);
+        break;
+      case MODIFIED:
+        modifyMessage(smsMessage);
+        break;
+      case REMOVED:
+        removeMessage(smsMessage);
+        break;
+    }
+  }
+
   public void addMessage(SmsMessage smsMessage) {
     messages.add(smsMessage);
-    if (messagesListener != null) {
-      messagesListener.messageAdded();
-    }
+    notifyListeners(getIndexById(smsMessage), smsMessage, Type.ADDED);
   }
 
   public void removeMessage(SmsMessage smsMessage) {
     int index = getIndexById(smsMessage);
     if (index != -1) {
-      messages.remove(smsMessage);
+      messages.remove(index);
       firestoreClient.deleteMessage(smsMessage);
-      if (messagesListener != null) {
-        messagesListener.messageRemoved(index);
-      }
-    } else {
-      Log.e(
-          TAG,
-          "Attempt to remove message by its ID, but message not found\nmessage id: "
-              + smsMessage.getId());
+      notifyListeners(index, smsMessage, Type.REMOVED);
```

Comment:
Why did you remove the error logging on not finding a message?

Reply:
Started a branch for adding logs for sensitive scenarios, removed this one now by mistake.
Added back this specific one. I'll push a PR with others later..